services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchRulesSpec.groovy

```diff
@@ -248,6 +249,81 @@ class SwitchRulesSpec extends BaseSpecification {
         ]
     }
 
+    def "Able to validate a switch in case flow is created with protected path"() {
+        given: "A switch and a flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+        def amountOfRules = { SwitchId switchId ->
+            (switchId == srcSwitch.dpId || switchId == dstSwitch.dpId) ? 3 : 4
+        }
+        assert northbound.getFlowPath(flow.id).protectedPath
+
+        when: "Validate rules on the switches"
+        def mainFlowPath = northbound.getFlowPath(flow.id).forwardPath
+        def protectedFlowPath = northbound.getFlowPath(flow.id).protectedPath.forwardPath
+        def commonNodeIds = mainFlowPath*.switchId.intersect(protectedFlowPath*.switchId)
+
+        then: "Rules are stored in the 'proper' section"
+        commonNodeIds.each { switchId ->
+            def rules = northbound.validateSwitchRules(switchId)
+            assert rules.properRules.size() == amountOfRules(switchId)
+            assert rules.missingRules.empty
+            assert rules.excessRules.empty
+        }
+
+        def uniqueNodes = protectedFlowPath.findAll { !commonNodeIds.contains(it.switchId) } + mainFlowPath.findAll {
+            !commonNodeIds.contains(it.switchId)
+        }
+        if (uniqueNodes){
+            uniqueNodes.each { sw ->
+                def rules = northbound.validateSwitchRules(sw.switchId)
+                assert rules.properRules.size() == 2
+                assert rules.missingRules.empty
+                assert rules.excessRules.empty
+            }
+        }
+
+        when: "Delete rule of protected path on the srcSwitch"
+        def protectedPath = northbound.getFlowPath(flow.id).protectedPath.forwardPath
+        def srcSwitchRules = northbound.getSwitchRules(commonNodeIds[0]).flowEntries.findAll {
+            !org.openkilda.model.Cookie.isDefaultRule(it.cookie)
+        }
+
+        def ruleToDelete = srcSwitchRules.find {
+            it.instructions?.applyActions?.flowOutput == protectedPath[0].inputPort.toString() &&
+                    it.match.inPort == protectedPath[0].outputPort.toString()
+        }.cookie
+
+        northbound.deleteSwitchRules(commonNodeIds[0], ruleToDelete)
+
+        then: "Deleted rule is moved to the 'missing' section on the srcSwitch"
+        def srcSwitchValidateRules = northbound.validateSwitchRules(commonNodeIds[0])
+        srcSwitchValidateRules.properRules.size() == 2
+        srcSwitchValidateRules.missingRules.size() == 1
```

Comment:
you can verify its cookie. I think it's better to do `srcSwitchValidateRules.missingRules == [ruleToDelete]` here